src/app/dataset-block/metadata-block/components/event-details/components/common/base-property/base-property.component.ts

```diff
@@ -0,0 +1,9 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
```

Comment:
@Component annotation is normally not required for abstract classes. It's enough that the concrete subclasses define it